### ISSUE TITLE
Simulator Halts Prematurely During Integration test

### ISSUE THREAD

**Issue by cameroncf** [bboure/serverless-appsync-simulator#72]
**Describe the bug**
I've been working on getting the Simulator up and running for some Integration testing w/Jest. I've roughly been following instructions found in a pretty neat little [article about doing this using serverless-offline](https://niradler.com/serverless-integration-tests-with-offline-plugin/). The goal is to run some Jest tests in my CI/CD pipeline, using node-fetch to automate posting queries and mutations against this simulator prior to deploy.

I can successfully start the simulator in my Jest beforeAll() and it will run the test all the way through with node-fetch simulating the client, then shut down the simulator when it's done using Jest's afterAll().

_However....._ 

I can run the same test, change no code, and it will succeed and fail randomly. This behavior appears to be due a race condition where sometimes the simulator shuts down before the test can run.

**To Reproduce**
I'm working on a more concise reproducible set of code. There's way too much noise in the current code to be useful here yet. I'll update this ticket once that happens but wanted to post anyway in case there is an obvious quick answer I am just not seeing.

**Expected behavior**
Success with Jest integration tests should be repeatable.

**Screenshots**
This is the console output around the simulator shutting down.
```
offline: Function names exposed for local invocation by aws-sdk:
[offline] Lambda Invocation Routes (for AWS SDK or AWS CLI):
[offline] Lambda Async Invocation Routes (for AWS SDK or AWS CLI):
AppSync Simulator: Halting AppSync Simulator
```

Then the jest test fails with error (sometimes):
`FetchError: request to http://192.168.1.39:20002/graphql failed, reason: connect ECONNREFUSED 192.168.1.39:20002`

**Additional context**
Commenting out this plugin's [binding to the 'before:offline:start:end' hook](https://github.com/bboure/serverless-appsync-simulator/blob/master/src/index.js#L34) makes my problem go away entirely and the tests runs correctly every time without fail.

**Comment by cameroncf** [comment]
I've isolated as much noise out of the code as possible and uploaded it into a [project called simtest](https://github.com/cameroncf/simtest). Note, the org and app names may need to be changed to your own if running this project.

Running `yarn dev` spins up the Amplify GraphQL Explorer and it runs fine in the browser.

Running `yarn test` will run 3 very basic and identical Jest tests. The first will succeed but the second and third fail because the AppSync Simulator mysteriously halts after the first one.

I cannot manage to get any debugging or logging on _WHY_ the simulator is halting, but it is.

If I can get this running, it will make me way more confident that code sent to AppSync is going to run properly. Hopefully someone working with or using this project will be willing to take a peek.

**Comment by cameroncf** [comment]
After dropping this problem and coming back to it a little while later I've discovered the cause of the problem. I'll put it here as some Search Engine Fodder in case someone else runs into this same problem.

Turns out that serverless-offline runs the start hook and then waits for SIGTERM or SIGINT because calling the end hook. However, it only waits for SIGTERM/SIGINT when `process.env.NODE_ENV !== 'test'`. ([relevant source here](https://github.com/dherault/serverless-offline/blob/master/src/ServerlessOffline.js#L99-L103))

In my case I am running this plugin as part of a jest test so it skips the wait and immediately terminates, which makes the jest testing go sideways most (but not all) of the time, depending on how fast each steps runs.

Resolution was to set environment to development in the spawned process like so:

```
slsOfflineProcess = spawn('sls offline start -s dev', {
    shell: true,
    env: { ...process.env, NODE_ENV: 'development' },
    cwd: process.cwd(),
  })
```

Seems to be working so far.
